src/engine/strat_engine/engine.rs

```diff
@@ -122,6 +154,39 @@ impl Engine for StratEngine {
     fn pools(&self) -> Vec<&Pool> {
         self.pools.into_iter().map(|x| x as &Pool).collect()
     }
+
+    fn get_eventable(&mut self) -> EngineResult<Option<Box<Eventable>>> {
+        Ok(Some(Box::new(DM::new()?)))
+    }
+
+    fn evented(&mut self) -> EngineResult<()> {
+        let device_list = DM::new()?.list_devices()?;
+
+        for &(ref dm_name, _, new_event_nr) in &device_list {
+            let new_event_nr = new_event_nr.expect("Supported DM versions always provide a value");
```

Comment:
Now that we know that ```new_event_nr``` is always ```Some``` this is simpler to think about.

I still think that

```
if new_event_nr > self.watched_dev_last_event_nrs.get(dm_name) {
  ...
}
```
is the correct thing to do.

There are only two possiblites.

1. ```prev_event_nr``` is ```None```. In that case, the device has shown up since the last time ```evented()``` was called. So there may be something to do about it, and we should handle it.
2. ```prev_event_nr``` matches ```Some```. In that case, there will be a comparsion between the values of the two numbers and if ```new_event_nr``` is greater than ```prev_event_nr``` something will be done.

Reply:
Doesn't compile.
```
error[E0308]: mismatched types
   --> src/engine/strat_engine/engine.rs:167:31
    |
167 |             if new_event_nr > self.watched_dev_last_event_nrs.get(dm_name) {
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected u32, found enum `std::option::Option`
    |
    = note: expected type `u32`
               found type `std::option::Option<&u32>`
```

Reply:
Yeah, you need to lose the ```expect()``` call as well. Then they are both ```Option``` type and hopefully ```Option``` has enough cleverness that ```>``` will still work.

Reply:
But in any case, if you do nothing, I think you have  a bug.